engine/table/src/main/java/io/deephaven/engine/table/impl/BaseTable.java

```diff
@@ -737,6 +745,13 @@ public final void notifyListeners(final TableUpdate update) {
             Assert.assertion(updateToSend.shifted().empty(), "updateToSend.shifted.empty()");
         }
 
+        final boolean hasNoListeners = !hasListeners();
+        if (hasNoListeners) {
+            lastNotificationStep = currentStep;
+            updateToSend.release();
+            return;
+        }
+
```

Comment:
We think this code is bad, because it's incorrect without synchronization. If we also intend to always do all of our validation even with no listeners, it's better to simply delete it.
```suggestion
```